### DISCUSSION TITLE
slow chunkserver optimisation tips [question]

### DISCUSSION THREAD

**Discussion by eleaner** [moosefs/moosefs#331] [Q&A]
Hi,
I have a few chunk servers utilising an elderly AMD Turion II Neo Dual-Core
(they are dedicated only to this task)
these are HP Microservers so can handle up to six HDD.

in the cases of rebalancing of the cluster (e.g few TB deletions, new chunk server), they get seriously stressed 
I am running one chunk server to cover all HDD. when I tried to run multiple chunk servers machine would become unresponsible

Are there any parameters I could tweak to make them operate better?
The "best practices" section on the website is pretty limited. Is there something more detailed somewhere? Examples of tweaking?

**Comment by chogata** [comment]
Hi,

First question - how much RAM do your chunk servers have? Does the system swap a lot? If it does, there is the first reason.

Second - multiple chunk servers on one machine are not an option in your case, stay with one process per machine. MooseFS has replication limits per chunk server, so when you multiplied the number of processes, you multiplied the number of replications, and your machines couldn't handle that.

To help alleviate the stress on your machines you can lower those replication limits in the master server configuration. Re-balancing will take more time then, but the whole system should be more responsive even during the re-balancing process.

You should change this setting in `mfsmaster.cfg`:

`CHUNKS_WRITE_REP_LIMIT = 2,1,1,4`

Lower the last value. You can start with 2, if it's still too high, change it to 1. This will limit the number of "new chunk server" re-balance replications.

**Comment by oszafraniec** [comment]
https://robo.moosefs.com/mfscalc/ - this can help you with how much RAM you need; as chogata wrote, swapping is a killer, if you have enough RAM even try ``swapoff`` it.

Rebalancing between chunkservers will always generate traffic but in my experience default values are OK if you have a lot CHSRVs (20 in my case). If you have only a few, traffic per CHSRV will be higher and yes there will be problems (been there, done that while having only f.ex. 3-5 CHSRVs in the cluster). And 6 HDDs is also not so much, better to have more of them so you have more IOPS per CHSRV.

If you deleted a lot of data and ended with CHSRVs internal rebalancing that takes lot of time, try "high speed rebalance" ( see #39 ) to make HDDs equal in used space fast to get rid of internal rebalancing state.

Best optimalization tip from me... Have as much of chunkservers as you can, than all operations propagates on them (CHSRVs talk to each other if the have to).

Hope this helps you :)
